### PR TITLE
Fixed static analyzer warnings

### PR DESCRIPTION
**Pull request by civanch**
The most of fixes are trivial and fixes are obvious. Few places may provide crash. For example, RunManagerMTWorker in abnormal case may crash at exit, which will shadow real problem.

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by smuzaffar** [comment]
Ok I have restarted the tests

**Comment by civanch** [comment]
+1
comparisons shows statistical agreement and not full identity, because part of fixes was removal of unused calls to random number generator.

File: SimTracker/Common/src/SiG4UniversalFluctuation.cc [hunk 1]
```diff
@@ -222,7 +207,7 @@ double SiG4UniversalFluctuation::SampleFluctuations(const double momentum,
   //
   if (suma > sumalim)
   {
-    p1 = 0., p2 = 0 ;
+    double p1, p2;
```
Comment (davidlange6) [inline SimTracker/Common/src/SiG4UniversalFluctuation.cc]: @civanch - you need the initialization here, no?

Comment (civanch) [inline SimTracker/Common/src/SiG4UniversalFluctuation.cc]: Hi David, no, because both p1 and p2 are defined below in any case. The static analyser complains to this line.

Comment (davidlange6) [inline SimTracker/Common/src/SiG4UniversalFluctuation.cc]: I see if (something) p1=blah then if (something2) p2=blah, then bah=p1*a+p2*b. So you definitely need to initialize p1 and p2.

Comment (civanch) [inline SimTracker/Common/src/SiG4UniversalFluctuation.cc]: HI David, the full block:
double p1, p2;
    if((a1+a2) > 0.)
    {
      // excitation type 1
      if (a1>alim) {
        siga=sqrt(a1) ;
        p1 = max(0., CLHEP::RandGaussQ::shoot(engine, a1, siga) + 0.5);
      } else {
	p1 = double(CLHEP::RandPoissonQ::shoot(engine,a1));
      }
    
      // excitation type 2
      if (a2>alim) {
        siga=sqrt(a2) ;
        p2 = max(0., CLHEP::RandGaussQ::shoot(engine, a2, siga) + 0.5);
      } else {
	p2 = double(CLHEP::RandPoissonQ::shoot(engine,a2));
      }
    
      loss = p1*e1Fluct+p2*e2Fluct;
 
      // smearing to avoid unphysical peaks
      if (p2 > 0.)
        loss += (1.-2.*CLHEP::RandFlat::shoot(engine))*e2Fluct;
      else if (loss>0.)
        loss += (1.-2.*CLHEP::RandFlat::shoot(engine))*e1Fluct;
      if (loss < 0.) loss = 0.0;
    }

so, for both p1 and p2 we have if()p1=value1; else p1=value2, the same for p2 - both are defined in any case.

However, what I see now: the line 'double p1, p2;' may be moved down inside the block. Minor thing, should I do this?

**Comment by civanch** [comment]
+1
additional commit adds a minor code improvements, also unused commented out lines of code are removed

**Comment by civanch** [comment]
please test

**Comment by davidlange6** [comment]
ah -whoops - thanks @civanch

**Comment by davidlange6** [comment]
+1
